### PR TITLE
Add command 'список заявок'

### LINKED ISSUE #41: Add command 'vacations list'
Add command 'vacations list' - list of leave requests sorted by status, date. 
Realization: can be used same sorting function as for 'birthdays list', so it needed to be moved to routines module as well.

### PR DESCRIPTION
**Pull request by BehindLoader**
close #41

### CODE REVIEW

File: src/viva.js [hunk 1]
```diff
@@ -432,6 +432,46 @@ module.exports = async (robot) => {
...
+    const formatLine = user => {
+      const username = user.name
+      const from = user.vivaLasVegas.leaveStart
+      const to = user.vivaLasVegas.leaveEnd
+
+      return ` @${username} ` +
+        `${from.day}.${from.month}.${from.year} - ` +
+        `${to.day}.${to.month}.${to.year}`
+    }
+
```
Comment (polina-popova) [inline src/viva.js]: Fix output date format: now is `13.9.2019 - 15.9.2019` --> `13.09.2019 - 15.09.2019`

File: src/viva.js [hunk 2]
```diff
@@ -432,6 +432,46 @@ module.exports = async (robot) => {
...
+      .join('\n')
+    const pending = users
+      .filter(user => user.vivaLasVegas && user.vivaLasVegas.requestStatus === PENDING_STATUS)
+      .sort(sorting)
+      .map(formatLine)
+      .join('\n')
+
+    const result = []
+
+    if (approved) result.push('*Принятые заявки:*\n' + approved)
```
Comment (polina-popova) [inline src/viva.js]: 'Одобренные заявки'

File: src/viva.js [hunk 3]
```diff
@@ -432,6 +432,46 @@ module.exports = async (robot) => {
...
+    if (pending) result.push('*Ожидающие подтверждения:*\n' + pending)
+    if (!result.length) result.push('Никто не собирается в отпуск')
+
```
Comment (polina-popova) [inline src/viva.js]: Dot in the end of the sentence.

File: src/viva.js [hunk 4]
```diff
@@ -432,6 +432,46 @@ module.exports = async (robot) => {
...
+      return ` @${username} ` +
+        `${from.day}.${from.month}.${from.year} - ` +
+        `${to.day}.${to.month}.${to.year}`
+    }
+
+    const sorting = (a, b) => {
+      const first = moment(`${a.vivaLasVegas.leaveStart.day}.${a.vivaLasVegas.leaveStart.month}.${a.vivaLasVegas.leaveStart.year}`, 'DD.MM.YYYY')
+      const second = moment(`${b.vivaLasVegas.leaveStart.day}.${b.vivaLasVegas.leaveStart.month}.${b.vivaLasVegas.leaveStart.year}`, 'DD.MM.YYYY')
+
```
Comment (polina-popova) [inline src/viva.js]: May be move `a.vivaLasVegas.leaveStart` and `b.vivaLasVegas.leaveStart.` to the variables.

File: src/viva.js [hunk 5]
```diff
@@ -432,6 +432,46 @@ module.exports = async (robot) => {
     msg.send('ok')
   })
 
+  robot.respond(/vacation list\s*/i, msg => {
```
Comment (polina-popova) [inline src/viva.js]: As in all viva-las-vegas script the command should be in Russian.

File: src/viva.js [hunk 6]
```diff
@@ -437,6 +437,57 @@ module.exports = async (robot) => {
...
+      .join('\n')
+    const pending = users
+      .filter(user => user.vivaLasVegas && user.vivaLasVegas.requestStatus === PENDING_STATUS)
+      .sort(sorting)
+      .map(formatLine)
+      .join('\n')
+
+    const result = []
+
+    if (approved) result.push('*Одобренные заявки:*\n' + approved)
```
Comment (eugulixes) [inline src/viva.js]: Use template string.

File: src/viva.js [hunk 7]
```diff
@@ -437,6 +437,57 @@ module.exports = async (robot) => {
...
+    if (pending) result.push('*Ожидающие подтверждения:*\n' + pending)
```
Comment (eugulixes) [inline src/viva.js]: Use template string.

File: src/viva.js [hunk 8]
```diff
@@ -437,6 +437,57 @@ module.exports = async (robot) => {
...
+      const username = user.name
+      const from = user.vivaLasVegas.leaveStart
+      const to = user.vivaLasVegas.leaveEnd
+
+      const formattedDate =
+        moment(`${from.day}.${from.month}.${from.year}`, 'D.M.YYYY').format('DD.MM.YYYY') +
+        ' - ' +
+        moment(`${to.day}.${to.month}.${to.year}`, 'D.M.YYYY').format('DD.MM.YYYY')
+
+      return ` @${username} ` + formattedDate
```
Comment (eugulixes) [inline src/viva.js]:
```
` @${username} ${formattedDate} `
```
Wouldn't be nicer?

**Pull request merged**